internal/upgrade/upgrade.go

```diff
@@ -36,14 +36,22 @@ func Run() {
 		})
 
 		It("has all the worker nodes running", func() {
-			values := &application.ClusterValues{}
-			err := state.GetFramework().MC().GetHelmValues(cluster.Name, cluster.GetNamespace(), values)
+			var valuesWithOldSchema *application.ClusterValues
+			valuesWithNewSchema := &common.ClusterValues{}
+			err := state.GetFramework().MC().GetHelmValues(state.GetCluster().Name, state.GetCluster().GetNamespace(), valuesWithNewSchema)
 			Expect(err).NotTo(HaveOccurred())
 
+			if common.ClusterIsUsingNewHelmValuesSchema(valuesWithNewSchema) {
```

Comment:
If cluster is using the new schema, convert new Helm values to old Helm values, so we don't have to change the rest of the code that is still needed for other providers other than CAPA.

Once all providers have been restructured, here we will revert the changes and use just the new struct for the new cluster app Helm values schema.